drivers/SmartThings/zigbee-bed/src/shus-mattress/init.lua

```diff
@@ -0,0 +1,422 @@
+-- Copyright 2024 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+local cluster_base = require "st.zigbee.cluster_base"
+local custom_clusters = require "shus-mattress/custom_clusters"
+local custom_capabilities = require "shus-mattress/custom_capabilities"
+
+local FINGERPRINTS = {
+  { mfr = "SHUS", model = "SX-1" }
+}
+
+-- #############################
+-- # Attribute handlers define #
+-- #############################
+
+local function process_switch_attr(device, value, cmd)
+  if value.value == false then
+    device:emit_event(cmd.off())
+  elseif value.value == true then
+    device:emit_event(cmd.on())
+  end
+end
+
+local function left_ai_mode_attr_handler(driver, device, value, zb_rx)
+  process_switch_attr(device, value, custom_capabilities.ai_mode.left)
+end
+
+local function right_ai_mode_attr_handler(driver, device, value, zb_rx)
+  process_switch_attr(device, value, custom_capabilities.ai_mode.right)
+end
+
+local function auto_inflation_attr_handler(driver, device, value, zb_rx)
+  process_switch_attr(device, value, custom_capabilities.auto_inflation.inflationState)
+end
+
+local function strong_exp_mode_attr_handler(driver, device, value, zb_rx)
+  process_switch_attr(device, value, custom_capabilities.strong_exp_mode.expState)
+end
```

Comment:
What we prefer to do in situations like this is have a factory function:

```
local function process_switch_attr_factory(cmd)
  return function(driver, device, value, zb_rx)
    if value.value == false then
      device:emit_event(cmd.off())
    ...
    end
  end
end
```

and then in the handlers, just write:

```
[custom_clusters.shus_smart_mattress.attributes.left_ai_mode.id] = process_switch_attr_factory(custom_capabilities.ai_mode_left,
```

Here's an example from our existing drivers: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-thermostat/src/init.lua#L186